test moving functionality to library

RavisankarSelvaraju opened the pull request:
Removed unnecessary variables and functions and moved them to the control-robot_frames library. This component can use that library to get the transformation to create dynamic ports at the end

Commit:
test moving functionality to library

maltewi commented on tasks/ChainPublisher.hpp:
```diff
@@ -21,22 +20,8 @@ class ChainPublisher : public ChainPublisherBase
-                       const std::vector<std::string>& involved_joints,
-                       KDL::JntArray& joint_array);
+     base::samples::Joints joint_state;
+     std::vector<RTT::OutputPort<base::samples::RigidBodyState>*> out_ports_;
```
I guess a member variable for the new `ChainTransformationCalculator` is missing here

RavisankarSelvaraju replied:
yeah, i'll will add that

Commit:
modify .orogen file to point to library for robot_frameTypes

Commit:
simplify component code for updated robot_frames library

maltewi commented on tasks/ChainPublisher.cpp:
```diff
@@ -83,92 +29,15 @@ bool ChainPublisher::configureHook()
-    }
+    ChainTransformationCalculator transform_calculator(chain_definitions, urdf_file_path);
+    // assign the created object to the chain_transformer pointer to access this object in update_hook 
+    chain_transformer = &transform_calculator;
```
You assign here a pointer of a temporary variable to a member. This will likely cause segfaults.

You'll need to create the instance of `ChainTransformationCalculator` with `new`. And don't forget to delete the pointer to prevent memory leak. The right place for deleting would be the clean up hook.

maltewi commented on tasks/ChainPublisher.hpp:
```diff
@@ -20,12 +20,16 @@ class ChainPublisher : public ChainPublisherBase
 {
     friend class ChainPublisherBase;
 protected:
+     std::vector<robot_frames::Chain> chain_;
```
this variabale is not used? remove it then

maltewi commented on tasks/ChainPublisher.hpp:
```diff
@@ -20,12 +20,16 @@ class ChainPublisher : public ChainPublisherBase
     friend class ChainPublisherBase;
 protected:
+     std::vector<robot_frames::Chain> chain_;
+     std::string urdf_file_;
```
this variable is not used? remove it then

maltewi commented on tasks/ChainPublisher.cpp:
```diff
@@ -83,92 +29,15 @@ bool ChainPublisher::configureHook()
     //Create dynamic output per chain
-    for(int i=0; i<n_defined_chains_; i++){
-        std::string port_name = chain_names_[i];
+    for(int i=0; i< chain_transformer->get_number_chains(); i++){
```
Can't you just iterate `chain_transformer->get_chain_names()` directly?

`for(const std::string& port_name :  chain_transformer->get_chain_names()){ ... }`

maltewi commented on tasks/ChainPublisher.cpp:
```diff
@@ -188,28 +57,12 @@ void ChainPublisher::updateHook()
-            bt_frames_[i].time = joint_state_.time;
+    while (_input.read(joint_state, false) == RTT::NewData){
+        chain_transformer->update_transforms(joint_state, bt_frames_);
+        for(int i=0; i< chain_transformer->get_number_chains(); i++){
```
Same as above... just iterate `bt_frames_` directly

Commit:
bug fix
fixed a pointer and deleting the pointer at cleanup
modify variable name
fix loop iteration

maltewi commented on tasks/ChainPublisher.cpp:
```diff
@@ -83,97 +29,18 @@ bool ChainPublisher::configureHook()
                 new RTT::OutputPort<base::samples::RigidBodyState>(port_name);
         ports()->addPort(port_name, *output_port);
-        out_ports_[i] = output_port;
+        output_ports_.push_back(output_port);
```
Here you extend the `output_ports`-vector without ensuring it is empty beforehand.
It will work fine for first time creation of the component. However, in Rock it is a normal use case to reconfigure tasks without destroying it (i.e. call stop, clean_up, apply_config, configure on a running task). In this case this would cause the output_ports_-vector to grow further with each reconfiguration.

At some point you should call `clear` on `output_ports_`. The right place would be cleanup-hook (or configure).

Btw.: I think the same applies to dynamically created ports. They should be removed in cleanup-hook. I think `ports()->removePort(pname)` is the function to call

Commit:
clear ports and vector for every configuration

maltewi reviewed (approved):
Needs https://github.com/rock-control/control-robot_frames/pull/4 to be merged
